client/src/Notifications.js

```diff
@@ -0,0 +1,125 @@
+            type: 'HOMEPAGE_ACCESS',
+            payload: result.data });
+        }
+        else {
+          console.log('Error: no result on mount.')
+          this.props.dispatch({ type: 'LOGOUT' });
+        }
+      })
+      .catch(error => {
+        console.log(error)
+        this.props.dispatch({ type: 'LOGOUT' });
+      })
+  }
+
+  render() {
+    const { cookies } = this.props;
+    console.log("notifications cookies:", cookies.get('token'),
+    this.props.logged_in);
+
+    if (this.props.loading){
+      return (<Loading component="Notifications"/>);
+    }
+    else if (!this.props.logged_in){
+      return (<Redirect to="/" />);
+    }
+    else if (Object.keys(this.props.user_info).length === 0){
+      return (<Loading component="Notifications"/>);
+    }
+    else {
+      console.log('CHECK',this.state.max_results, this.props.user_info.notifications.length)
+      return (
+      <div style={{padding:'2em',paddingTop:'0', 
+        display:'flex', justifyContent:'center'}}>
+      <Jumbotron style={{minHeight:'27em', paddingTop:'0', 
+        textAlign:'center'}}>
+        <div>
+          <h4>Notifications {' '}
+          {this.props.user_info.unread_notifications_count > 0 ? 
+          <Badge color="primary">
+            {this.props.user_info.unread_notifications_count}
+          </Badge>
+          : null}
+          </h4>
+        </div>
+        
+        {Object.keys(this.props.user_info.notifications).reverse().map(
+          function(key) {
+          return(
+            (this.props.user_info.notifications.length > this.state.max_results && 
+              key >= this.props.user_info.notifications.length - this.state.max_results)
+              || this.props.user_info.notifications.length <= (this.state.max_results) ? 
+            <Toast key={key} style={{width:'40em'}}>
+            <ToastBody style={{fontSize:'13px', paddingLeft:'0',
+                backgroundColor: this.props.user_info.notifications[key].read ? 
+                'lightgray':'white', paddingBottom:'0', 
+                paddingTop:'0.3em'}}>
+
+              <Notification 
+                is_main_page={true}
+                token={this.props.cookies.get('token')}
+                dispatch={this.props.dispatch} index={key} 
+                read={this.props.user_info.notifications[key].read}
+                content={this.props.user_info.notifications[key].content}/>
+
+              <hr className="my-6" 
+                style={{marginTop:'0.3em', marginBottom:'0'}}/>
+            </ToastBody>
+            </Toast> : null
+          )
+        }.bind(this))}
+
+        {(this.state.max_results >= this.props.user_info.notifications.length) 
+          ? null :
+        <div style={{display:'flex', justifyContent:'center', 
+          paddingTop:'0.5em', paddingBottom: '4em'}}>
+          <Button color="light" size="sm"
+            onMouseEnter={()=>{
+              this.setState({max_results: this.state.max_results + 10})

```

Comment:
might be easier to make a single variable that holds the page size instead of specifying it like this. this will make changing the page size in the future easier

Reply:
Was thinking this, decided to not do that for now because it only comes up twice and would take up a lot more space (this.state.page_size vs 10), and the 10 magic number would still have to be used instead of the variable for the initialization of the max_results state variable. Thoughts?

Reply:
i think that we should end up having pagination for multiple parts of the app (posts, post search, comments are a few I can think of right now) so it could be worth it to have a separate library or helper functions that handle pagination.

it could look something like `paginate(list shown, list not_shown, int page_size)` that returns an object with the new `shown` and `not_shown` lists along with a boolean that is true if there are objects in `not_shown` after the new page is loaded.

either way i think it would be better to store the page size in a separate variable even if it will take more space in favor of only having to change one variable in order to change page size. we should be able to initialize `max_results` using the new static variable that we create